automates/model_assembly/networks.py

```diff
@@ -270,18 +272,21 @@ def __eq__(self, other) -> bool:
         return (
             self.occurrence_num == other.occurrence_num
             and self.border_color == other.border_color
-            and set([n.uid for n in self.nodes])
-            == set([n.uid for n in other.nodes])
+            and set([n.uid for n in self.nodes]) == set([n.uid for n in other.nodes])
         )
 
     @classmethod
-    def from_container(cls, con: GenericContainer, occ: int):
+    def from_container(
+        cls, con: GenericContainer, occ: int, parent_subgraph: "GrFNSubgraph"
```

Comment:
@dpdicken why is the type annotation for `parent_subgraph` the string `"GrFNSubgraph"`? Is there some reason why we can't use the class name itself like we did for `con`?

Reply:
When doing references to the class that its contained within, you have to put it as a string literal. Another option is to add `from __future__ import annotations` as an import and the module will automatically convert all type hints to strings for you so you dont have to manually do it in these situations.  They are supposed to handle this in python 3.10. 

(I got this info from here :) https://stackoverflow.com/questions/33533148/how-do-i-type-hint-a-method-with-the-type-of-the-enclosing-class)

Reply:
@dpdicken thanks for the pro-tip! Let's go ahead and use the future annotations import.

Reply:
I went ahead and did the update that should resolve this conversation.